code/game/gamemodes/dynamic/dynamic_rulesets.dm

```diff
@@ -162,6 +164,11 @@
 		return FALSE
 	return TRUE
 
+/// Runs from gamemode process() if ruleset fails to start, like delayed rulesets not getting valid candidates.
+/// This one only handles refunding the threat, override in ruleset to clean up the rest.
+/datum/dynamic_ruleset/proc/clean_up()
+	mode.refund_threat(cost + (scaled_times * scaling_cost))
```

Comment:
Threat log should be used, something like:
`threat_log += "[worldtime2text()]: [ruletype] [name] refunded [cost + (scaled_times * scaling_cost)]"
`

Reply:
perhaps hook up the threat logging to the threat use/refund procs.